### DISCUSSION TITLE
Enable receiving intents for Tasker integration

### DISCUSSION THREAD

**Discussion by aliicex** [mullvad/mullvadvpn-app#8924] [Feature requests & Ideas]
It'd be really handy to be able to bring a WG tunnel up/down through an intent so that the Android app can be used with Tasker.

For example, [Viscerion](https://github.com/msfjarvis/viscerion/wiki/Tasker-integration) allows me to send
`me.msfjarvis.viscerion.debug.SET_TUNNEL_UP` or
`me.msfjarvis.viscerion.debug.SET_TUNNEL_DOWN`

**Comment by faern** [comment]
Our main purpose is to keep the user safe. I'm quite skeptical towards allowing any app to bring the tunnel down and allowing the phone to leak packets outside of the tunnel. But we'll discuss it.

**Comment by aliicex** [comment]
Thanks. One of the reasons I like you guys is that you don't just blindly implement things "because they're nice features", but put your threat model first.

Hopefully there's a good way to make this work. E.g. I can do something like this in bash on macOS to bring a tunnel up/down with tunnels which are present in the official Wireguard macOS app (though I could do something similar using the Mullvad CLI, I guess):
`scutil --nc stop "tunnelName"`
`scutil --nc start "tunnelName"`

thanks, again!

**Comment by JeffHenson** [comment]
+1 for the feature request.

My use case would be to use Tasker to automatically enable the tunnel when I connect to non-whitelisted wifi networks. That would make it a easier to use untrusted wifi connections knowing that the tunnel would automatically connect.

**Comment by aliicex** [comment]
That's exactly my use case, too. Could either be achieved through Tasker, or by being able to set an SSID whitelist/blacklist in the app itself; [linked issue](https://github.com/mullvad/mullvadvpn-app/issues/783)

**Comment by rperryng** [comment]
I too was hoping to build some tasker automation with mullvad based on the wifi neftwork connected.

I understand the need for putting privacy first, thanks for that :).

Could certain packages (like tasker's) be allow-listed to send intents to control mullvad (is this possible?  It's been a long time since I've done android development... 😅)?

**Comment by somerandoname** [comment]
Are there any news on this issue? The iOS app did support commands for some time through Shortcuts before running into [technical problems](https://github.com/mullvad/mullvadvpn-app/pull/4185).

It's already possible to automate connections with the native WireGuard app, but features like DAITA or split tunneling can't be used that way. So intents would still be a welcome addition to the Mullvad app.

**Comment by Pururun** [comment]
> Are there any news on this issue? The iOS app did support commands for some time through Shortcuts before running into [technical problems](https://github.com/mullvad/mullvadvpn-app/pull/4185).
> 
> It's already possible to automate connections with the native WireGuard app, but features like DAITA or split tunneling can't be used that way. So intents would still be a welcome addition to the Mullvad app.

I have created an internal issue for this, but spontaneously allowing any other app to start and stop the tunnel sounds really dangerous. Maybe if you could define a whitelist of apps that are allowed to manipulate Mullvad in this way?

**Comment by martijnarts** [comment]
What about allowing an external app to ask Tailscale to prompt for starting/stopping in a notification?

**Comment by Pururun** [comment]
> What about allowing an external app to ask Tailscale to prompt for starting/stopping in a notification?

That could be an idea, I will add it to the issue. :)
